### ISSUE TITLE
Time dependent Hamiltonian in mcsolve and mesolve product different result

### ISSUE THREAD

**Issue by kwyip** [qutip/qutip#490]
Using Time dependent Hamiltonian in mcsolve and mesolve product different result for expectation value for `sigmay()`. It seems to work well for expectation value for `sigmax()` or  `sigmaz()`. I wonder if dealing with complex number leads to this or there is another issue.

This is a simple code illustrating the problem.
```
from qutip import *
from scipy import *
from pylab import*
#sigmay problem

psi0 = (basis(2,0) + basis(2,1)).unit()   
rho0 = psi0*psi0.dag()

H0 = 0*sigmax()
H1 = sigmax()
print(psi0)
print(H0*psi0)
H2 = sigmaz()
def H1_coeff(t,args):
    return t-1

def H2_coeff(t,args):
    return -t

h_t = [H0,[H1, H1_coeff],
       [H2, H2_coeff]]

ntraj = 500

tlist = linspace(0, 3, 100)
medata = mesolve(h_t, rho0, tlist, [], [sigmay()], args = {})
mcdata = mcsolve(h_t, psi0, tlist, [], [sigmay()], ntraj = ntraj, args = {})


fig,axes = subplots(1,1,figsize=(12,6))
axes.plot(mcdata.expect[0], color="green", alpha =0.6, label = "mcsolve")
axes.plot(medata.expect[0], color="blue", alpha =0.6, label = "mesolve")
axes.set_xlabel(r'$\frac{t}{t_f}$')
axes.set_ylabel('Expectation value')
axes.legend(("mcsolve", "mesolve"))

show()
```

**Comment by nonhermitian** [comment]
Should be fixed in Pull #492.  Thanks for finding this.

**Comment by nonhermitian** [comment]
This was fixed in #492.
